src/models.zig

```diff
@@ -149,6 +161,296 @@ pub const Model = struct {
+				} else {
+					var coordsIter = std.mem.split(u8, line[2..], " ");
+					var faceData: [3][4]usize = undefined;
+					var i: usize = 0;
+					while (coordsIter.next()) |vertex| : (i += 1) {
+						var d = std.mem.split(u8, vertex, "/");
+						var j: usize = 0;
+						while (d.next()) |value| : (j += 1) {
+							faceData[j][i] = try std.fmt.parseUnsigned(usize, value, 10) - 1;
+						}
+					}
+					try quadFaces.append(.{.vertex=faceData[0], .uvs=faceData[1], .normals=faceData[2]});
+				}
+			}
+		}
+
+		var quadInfos = std.ArrayList(QuadInfo).init(main.stackAllocator.allocator);
+		defer quadInfos.deinit();
+
+		for (tris.items) |face| {
+			const normal: Vec3f = normals.items[face.normals[0]];
+			
+			var uvA: Vec2f = uvs.items[face.uvs[0]];
+			var uvB: Vec2f = uvs.items[face.uvs[1]];
+			var uvC: Vec2f = uvs.items[face.uvs[2]];
+
+			
+			var minUv = @floor(@min(@min(uvA, uvB), uvC));
+			const textureSlot = @as(u32, @intFromFloat(@floor(minUv[1]))) * 4 + @as(u32, @intFromFloat(@floor(minUv[0])));
+
+			uvA -= minUv;
+			uvB -= minUv;
+			uvC -= minUv;
+			
+			minUv = @min(@min(uvA, uvB), uvC);
+			const maxUv = @max(@max(uvA, uvB), uvC);
+
+			uvA[1] = (minUv[1] + maxUv[1]) - uvA[1];
+			uvB[1] = (minUv[1] + maxUv[1]) - uvB[1];
+			uvC[1] = (minUv[1] + maxUv[1]) - uvC[1];
+
+			const cornerA: Vec3f = vertices.items[face.vertex[0]];
+			const cornerB: Vec3f = vertices.items[face.vertex[1]];
+			const cornerC: Vec3f = vertices.items[face.vertex[2]];
+			
+			try quadInfos.append(.{
+				.normal = normal,
+				.corners = .{cornerA, cornerB, cornerC, cornerB},
+				.cornerUV = .{uvA, uvB, uvC, uvB},
+				.textureSlot = textureSlot,
+			});
+		}
+
+		for (quadFaces.items) |face| {
+			const normal: Vec3f = normals.items[face.normals[0]];
+			
+			var uvA: Vec2f = uvs.items[face.uvs[0]];
+			var uvB: Vec2f = uvs.items[face.uvs[1]];
+			var uvC: Vec2f = uvs.items[face.uvs[2]];
+			var uvD: Vec2f = uvs.items[face.uvs[3]];
+
+			var minUv = @floor(@min(@min(uvA, uvB), @min(uvC, uvD)));
+			const textureSlot = @as(u32, @intFromFloat(@floor(minUv[1]))) * 4 + @as(u32, @intFromFloat(@floor(minUv[0])));
+
+			uvA -= minUv;
+			uvB -= minUv;
+			uvC -= minUv;
+			uvD -= minUv;
+			
+			minUv = @min(@min(uvA, uvB), @min(uvC, uvD));
+			const maxUv = @max(@max(uvA, uvB), @max(uvC, uvD));
+
+			uvA[1] = (minUv[1] + maxUv[1]) - uvA[1];
+			uvB[1] = (minUv[1] + maxUv[1]) - uvB[1];
+			uvC[1] = (minUv[1] + maxUv[1]) - uvC[1];
+			uvD[1] = (minUv[1] + maxUv[1]) - uvD[1];
+
+			std.log.info("{any} {any}\n", .{minUv[1], maxUv[1]});
+
+			const cornerA: Vec3f = vertices.items[face.vertex[0]];
+			const cornerB: Vec3f = vertices.items[face.vertex[1]];
+			const cornerC: Vec3f = vertices.items[face.vertex[2]];
+			const cornerD: Vec3f = vertices.items[face.vertex[3]];
+			
+			try quadInfos.append(.{
+				.normal = normal,
+				.corners = .{cornerB, cornerA, cornerC, cornerD},
+				.cornerUV = .{uvB, uvA, uvD, uvC},
```

Comment:
This also looks wrong.
First of all why is it not ABCD?
Secondly why is it BACD for the corners, but BADC for the UVs? Maybe this explains why your UVs were wrong all the time?